src/smpi/smpi_shared.cpp

```diff
@@ -125,6 +127,7 @@ std::map<void*, shared_metadata_t> allocs_metadata;
 xbt_dict_t calls = nullptr;           /* Allocated on first use */
 #ifndef WIN32
 static int smpi_shared_malloc_bogusfile           = -1;
+static int smpi_shared_malloc_bogusfile_huge_page  = -1;
```

Comment:
Why do you need another global variable `smpi_shared_malloc_bogusfile_huge_page`? Can't you use  the `smpi_shared_malloc_bogusfile` variable in any case, regardless of whether we use huge pages or regular ones?